src/Service/ExperienceService.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+namespace App\Service;
+
+use App\DTO\ExperienceCollection;
+use App\Entity\Experience;
+use App\Entity\Host;
+use App\Repository\CategoryRepository;
+use App\Repository\ExperienceRepository;
+use App\Repository\HostRepository;
+use App\Request\ExperienceRequest;
+use JetBrains\PhpStorm\ArrayShape;
+
+class ExperienceService
+{
+    #[ArrayShape(['data' => "array", 'status' => "bool", 'message' => "string"])]
+    public function getAll(ExperienceRepository $repository, Host $host): array
+    {
+        $res = [];
+        $experiences = $repository->findBy(['host' => $host]);
+        if (!$experiences) {
+            $res['data'] = [];
+        }
+        foreach ($experiences as $experience) {
+            $experienceCollection = new ExperienceCollection();

```

Comment:
insead of using `set` and `get` functions, you can define variables as public field and use:
`$experienceCollection->id = $experience->getId()`.
But I really don't know which one is better! :)